### PR TITLE
System tray updates for Windows

### PR DESCRIPTION
**Pull request by atavism**
Resolves https://github.com/getlantern/engineering/issues/1416

Moves code for managing system tray to separate TrayHandler class and some other improvements

### CODE REVIEW

**Comment by atavism** [comment]
I tested on Windows and confirmed it resolves the issue https://github.com/getlantern/engineering/issues/1416

File: lib/common/tray_handler.dart [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+        ? 'assets/images/lantern_connected_32.ico'
+        : 'assets/images/lantern_connected_32.png';
+  }
+  return Platform.isWindows
+      ? 'assets/images/lantern_disconnected_32.ico'
+      : 'assets/images/lantern_disconnected_32.png';
+}
+
+
+class TrayHandler with TrayListener {
```
Comment (jigar-f) [inline lib/common/tray_handler.dart:20]: I like having a separate class for this, thanks for making changes.

File: lib/common/tray_handler.dart [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+      ],
+    );
+    await trayManager.setContextMenu(menu);
+  }
+
+  @override
+  void onTrayMenuItemClick(MenuItem menuItem) {
+    switch (menuItem.key) {
+      case 'show':
+        windowManager.show();
```
Comment (jigar-f) [inline lib/common/tray_handler.dart:59]: @atavism I think we should use `windowManager.focus();` instead of  `windowManager.show();`

**Review by jigar-f — COMMENTED** [review]
@atavism Just a couple of comments otherwise LGTM

File: lib/common/tray_handler.dart (hunk 2 above)
Comment (atavism) [reply lib/common/tray_handler.dart:59]: Not sure I understand the difference (show is supposed to show and give focus on the window) but I can change it

Comment (jigar-f) [reply lib/common/tray_handler.dart:59]: So if you have multiple windows open on Mac then if you click to show it does not show up foreground, I think we talked about this before, when I working on some desktop fixes, that's why I remember,

**Comment by jigar-f** [comment]
@atavism Looks good on macOS.

**Comment by atavism** [comment]
> Looks good on macOS.

Cool, thanks! Merging then..

**Pull request merged**